modules/tinymce/src/core/test/ts/browser/DragDropOverridesTest.ts

```diff
@@ -8,198 +8,288 @@ import { assert } from 'chai';
+      await Waiter.pWait(1500); // Wait a small amount of time to ensure the scrolling happens
+      assert.isAbove(editor.getWin().scrollX, initialScrollX); // Make sure scrolling happened
+    });
 
-  it('TINY-8874: Dragging CEF element towards the left edge causes scrolling', async () => {
-    const editor = hook.editor();
-    await Waiter.pWait(200); // Wait a small amount of time to ensure the events have been bound
-    editor.setContent(`
+    it('TINY-8874: Dragging CEF element towards the left edge causes scrolling', async () => {
+      const editor = hook.editor();
+      editor.setContent(`
       <div style="display: flex">
       <p class="target" style="margin-right: 16px">CEF can get dragged before this one</p>
       <p id="separator" style="flex: 0 0 200px;"></p>
       <p contenteditable="false" style="flex: 0 0 200px; background-color: black; color: white">Draggable CEF</p>
       </div>
     `);
-    editor.getBody().ownerDocument.defaultView.scroll({
-      left: 450
+      editor.getWin().scroll({
+        left: 450
+      });
+      const target = UiFinder.findIn(TinyDom.body(editor), 'p:contains("Draggable CEF")').getOrDie();
+      const initialScrollX = editor.getWin().scrollX;
+      Mouse.mouseDown(target);
+      Mouse.mouseMoveTo(SugarElement.fromDom(editor.getBody()), 2, 9); // Move the mouse close to the right edge of the editor to trigger scrolling
+      await Waiter.pWait(1500); // Wait a small amount of time to ensure the scrolling happens
+      Mouse.mouseUp(target);
+      assert.isBelow(editor.getWin().scrollX, initialScrollX); // Make sure scrolling happened
+    });
+  });
+
+  context('Tests with autoresize', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      indent: false,
+      menubar: false,
+      base_url: '/project/tinymce/js/tinymce',
+      height: window.innerHeight + 1000,
+      width: window.innerWidth + 1000,
+    }, [], true);
+
+    it('TINY-8874: Dragging CEF element towards the bottom edge causes scrolling when autoresize is set', async () => {
+      const editor = hook.editor();
+      await Waiter.pWait(500);
+      editor.setContent(`
+      <p contenteditable="false" style="height: 200px; background-color: black; color: white">Draggable CEF</p>
+      <p id="separator" style="height: 100px"></p>
+      <p class="hidden" style="height: 1500px"></p>
+      <p>CEF can get dragged after this one</p>
+    `);
+      const target = UiFinder.findIn(TinyDom.body(editor), 'p:contains("Draggable CEF")').getOrDie();
+      const dest = UiFinder.findIn(TinyDom.body(editor), 'p.hidden').getOrDie();
+      const initialScrollY = window.scrollY;
+      Mouse.mouseDown(target);
+      Mouse.mouseMoveTo(dest, 0, window.innerHeight - 2); // Move the mouse close to the bottom edge of the editor to trigger scrolling
+      await Waiter.pWait(1000); // Wait a small amount of time to ensure the scrolling happens
```

Comment:
The problem with these types of waits is that they slow down the whole testing. This entire test file now just waits for 10+ seconds. I think it's better to use a waiter here. Wait for the scroll to happen at intervals. Then it will hammer say every 10 ms to check did it scroll. That way it's likely to pass the test after 100ms instead of 1000 this goes for all these tests that has wait for a while better wait for a state to happen. If we really want to test that something happens after 100 ms then it could validate that the time before and after is more than 100 ms in a separate state and test. So use agars `Waiter.pTryUntil` or `Waiter.pTryUntilPredicate`. Maybe even create some form of pWaitForScrollX/pWaitForScrollY functions that does some form of setup, grabs the initial value, and a done. Since there seems to be a lot of repetition here but not sure that makes it easier to read the tests.

Reply:
Done! I used `Waiter.pTryUntil`